### ISSUE TITLE
Workspace symbol not populated when trying to use this Language server from 'neovim'

### ISSUE THREAD

**Issue by PrajitP** [richterger/Perl-LanguageServer#17]
# Note
**I don't think this is bug. It is mostly me trying to figure out how to configure this language server to work with 'vim' or 'neovim'.**

Given that all the code exist and is also being in used, and is tested with `vscode`. I am trying to understand how `vscode` language client works, specially when are method related to `workspace` get trigger and what is the content of those methods. 

Once I figure this out I will configure my `neovim` language client to send that command and also document it for others.

# TODO
* [ ] Configure this to work with [Neovim](https://neovim.io/) and add the documentation for it.
 
# How to reproduce
* My 'neovim rc' settings
```vim
call plug#begin('~/.config/nvim/plugged')
    Plug 'autozimu/LanguageClient-neovim', {
         \ 'branch': 'next',
         \ 'do': 'bash install.sh',
         \ }
call plug#end()

let g:LanguageClient_serverCommands = {
    \ 'perl': ['perl', '-MPerl::LanguageServer', '-e', 'Perl::LanguageServer::ru
n', '--', '--debug', '--version', '2.1.0'],
    \ }
let g:LanguageClient_loggingFile = expand('~/LanguageClient.log')
let g:LanguageClient_serverStderr = expand('~/LanguageClientStderr.log')
let g:LanguageClient_loggingLevel='DEBUG'
```
* Running the server on Perl code
```perl
my $string = "This is a test";
print $string, "\n";
```

* Vim command to trigger definition search,  when cursor is on `$string` on 2nd line.
```
:call LanguageClient#textDocument_definition()
```
result
```
Not found!
```

# Issue
After debugging found that the, `symbols` attribute for `$workspace`(`Perl::LanguageServer::Workspace`) are not populated. As a result of it, 2 of functionality `definitionProvider` and `referencesProvider` which depends on 'workspace symbols' don't work. Looking through code I found that in module `Perl::LanguageServer` there is a method
```perl
sub parsews {
    ...
}
``` 
which populates the symbols for workspace, but looks like it not called from anywhere in the code.

Module `Perl::LanguageServer::Methods::workspace` also has 2 method
```perl
sub _rpcnot_didChangeConfiguration {
    ...
}

sub _rpcnot_didChangeWorkspaceFolders {
    ...
}
```
were `symbols` for workspace are populated, I am trying to figure out how and when to trigger them form 'neovim client'.

**Comment by PrajitP** [comment]
It turn out [LanguageClient-neovim](https://github.com/autozimu/LanguageClient-neovim) I am using [does not support](https://github.com/autozimu/LanguageClient-neovim/issues/177) [workspace/workspaceFolders](https://microsoft.github.io/language-server-protocol/specification#workspace_workspaceFolders) feature. 

`Perl::LanguageServer::Parser->background_parser()` only parse workspace from 'workspace folder'.

Not sure if it's write design, but I think it should also parse 'rootURI' directory as it is consider as default root if 'workspace folder' is missing.

**Comment by richterger** [comment]
Fixed in commit 19721b14f8dbd1f0c84d7dee6d11f82dbdcb3157
